client/script.js

```diff
@@ -356,146 +243,217 @@ function draw_ships(stX, stY, size, brd, is_shot) {
+    }
+    switch (state) {
+        case -1:
+            draw_top(field_startX + 190, field_startY - 20);
+            break;
+        case 1:
+            draw_field(field_startX, field_startY);
+            draw_ships_count();
+            draw_button(field_startX + field_size + 70, field_startY + 15, "Auto", "#FFD500");
+            draw_nick();
+            break;
+        case 2:
+            draw_field(field_startX, field_startY);
+            draw_top(active_startX + field_size + 160, active_startY + 40);
+            draw_nick();
+            break;
+        case 3:
+            draw_field(field_startX, field_startY);
+            draw_field(active_startX, active_startY);
+            draw_top(active_startX + field_size + 160, active_startY + 40);
+            draw_nick();
+            if (opp_board) draw_ships(active_startX, active_startY, field_size, opp_board, true);
+            if (board_hits) draw_ships(field_startX, field_startY, field_size, board_hits, true);
+            break;
+    }
+};
+
+for (let i = 0; i < joinBtns.length; i++)
+    joinBtns[i].addEventListener("click", () => {
+        socket.send(
+            JSON.stringify(
+                i === 1
+                    ? {
+                          type: "join",
+                          data: i,
+                          code: codeInput.value
+                      }
+                    : {
+                          type: "join",
+                          data: i
+                      }
+            )
+        );
+    });
+
+form.addEventListener("submit", e => {
+    e.preventDefault();
+    const lgn = form.login.value;
+    const pswgd = form.password.value;
+
+    if (pswgd.length > 5 && lgn.length > 3 && pswgd.length <= 50 && lgn.length <= 20) {
+        if (lgn.match(/^[0-9a-zA-Z_-]+$/)) {
+            if (registr) return socket.send(JSON.stringify({ type: "authorization", method: "register", login: lgn, password: pswgd }));
+            return socket.send(JSON.stringify({ type: "authorization", method: "login", login: lgn, password: pswgd }));
+        }
+        return (loginAlert.textContent = "Please use only A-Z 0-9 characters");
+    }
+    loginAlert.textContent = "Login must be 4-20 characters long, password - 6-50";
+});
+
+logregBtn.addEventListener("click", () => {
+    loginBtn.textContent = registr ? "Log in" : "Sign up";
+    logregBtn.textContent = registr ? "Register" : "Log in";
+    logregText.textContent = registr ? "Already have an account? " : "Don't have an account? ";
+    registr = !registr;
+});
+
+redraw();
+
+socket.onopen = () => {
     console.log("Connected.");
-    token = localStorage.getItem('skrepka.battleships.token.login');
+    const token = localStorage.getItem("skrepka.battleships.token.login");
     if (token) {
-        socket.send(JSON.stringify({type: "authorization", method: "token", token: token}));
+        socket.send(JSON.stringify({ type: "authorization", method: "token", token }));
```

Comment:
Why?

Reply:
JS automatically converts {value} to {"value": value}. {token: token} was just unnecessary. {token} is more elegant and uses less characters.